homeassistant/components/assist_satellite/entity.py

```diff
@@ -248,31 +254,48 @@ async def async_accept_pipeline_from_satellite(
         # Set entity state based on pipeline events
         self._run_has_tts = False
 
-        await async_pipeline_from_audio_stream(
-            self.hass,
-            context=self._context,
-            event_callback=self._internal_on_pipeline_event,
-            stt_metadata=stt.SpeechMetadata(
-                language="",  # set in async_pipeline_from_audio_stream
-                format=stt.AudioFormats.WAV,
-                codec=stt.AudioCodecs.PCM,
-                bit_rate=stt.AudioBitRates.BITRATE_16,
-                sample_rate=stt.AudioSampleRates.SAMPLERATE_16000,
-                channel=stt.AudioChannels.CHANNEL_MONO,
-            ),
-            stt_stream=audio_stream,
-            pipeline_id=self._resolve_pipeline(),
-            conversation_id=self._conversation_id,
-            device_id=device_id,
-            tts_audio_output=self.tts_options,
-            wake_word_phrase=wake_word_phrase,
-            audio_settings=AudioSettings(
-                silence_seconds=self._resolve_vad_sensitivity()
+        self._pipeline_task = self.hass.async_create_background_task(
+            async_pipeline_from_audio_stream(
+                self.hass,
+                context=self._context,
+                event_callback=self._internal_on_pipeline_event,
+                stt_metadata=stt.SpeechMetadata(
+                    language="",  # set in async_pipeline_from_audio_stream
+                    format=stt.AudioFormats.WAV,
+                    codec=stt.AudioCodecs.PCM,
+                    bit_rate=stt.AudioBitRates.BITRATE_16,
+                    sample_rate=stt.AudioSampleRates.SAMPLERATE_16000,
+                    channel=stt.AudioChannels.CHANNEL_MONO,
+                ),
+                stt_stream=audio_stream,
+                pipeline_id=self._resolve_pipeline(),
+                conversation_id=self._conversation_id,
+                device_id=device_id,
+                tts_audio_output=self.tts_options,
+                wake_word_phrase=wake_word_phrase,
+                audio_settings=AudioSettings(
+                    silence_seconds=self._resolve_vad_sensitivity()
+                ),
+                start_stage=start_stage,
+                end_stage=end_stage,
             ),
-            start_stage=start_stage,
-            end_stage=end_stage,
+            name=f"{self.name}_pipeline",
         )
 
+        try:
+            await self._pipeline_task
+        finally:
+            self._pipeline_task = None
+
+    async def _cancel_running_pipeline(self) -> None:
```

Comment:
Doesn't have to be async.
```suggestion
    @callback
    def _async_cancel_running_pipeline(self) -> None:
```

Reply:
I await the cancelled task to ensure it's received the run end even, so it does need to be async.